pkg/proxy/conntrack/cleanup.go

```diff
@@ -122,22 +123,24 @@ func CleanStaleEntries(ct Interface, ipFamily v1.IPFamily,
 		origDst := entry.Forward.DstIP.String()   // match Service IP
 		origPortDst := int(entry.Forward.DstPort) // match Service Port
 		origPortDstStr := strconv.Itoa(origPortDst)
-		replySrc := entry.Reverse.SrcIP.String() // match Serving Endpoint IP
+		replySrc := entry.Reverse.SrcIP.String()   // match Serving Endpoint IP
+		replyPortSrc := int(entry.Reverse.SrcPort) // match Serving Endpoint Port
+		replyPortSrcStr := strconv.Itoa(replyPortSrc)
 
 		// if the original destination (--orig-dst) of the entry is service IP (ClusterIP,
 		// LoadBalancerIPs or ExternalIPs) and (--orig-port-dst) of the flow is service Port
 		// and the reply source (--reply-src) is not IP of any serving endpoint, we clear the entry.
```

Comment:
comment is now wrong because it doesn't mention replyPortSrc.
you could also simplify it: "if origDst:origPortDst is a service IP:port, and replySrc:replyPortSrc is not a serving endpoint"

(Also, I gotta agree with Adrian: conntrack's naming convention is weird. It's not so terrible that you _have_ to change it, but I definitely wouldn't object if it you did.)